src/main/java/org/spongepowered/common/entity/player/tab/SpongeTabList.java

```diff
@@ -0,0 +1,232 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.entity.player.tab;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Objects;
+import com.google.common.collect.Maps;
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.network.play.server.S38PacketPlayerListItem;
+import net.minecraft.network.play.server.S47PacketPlayerListHeaderFooter;
+import net.minecraft.util.ChatComponentText;
+import net.minecraft.util.IChatComponent;
+import org.spongepowered.api.entity.living.player.gamemode.GameMode;
+import org.spongepowered.api.entity.living.player.tab.TabList;
+import org.spongepowered.api.entity.living.player.tab.TabListEntry;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.common.interfaces.network.play.server.IMixinS38PacketPlayerListItem;
+import org.spongepowered.common.text.SpongeTexts;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import javax.annotation.Nullable;
+
+public final class SpongeTabList implements TabList {
+
+    private static final IChatComponent EMPTY_COMPONENT = new ChatComponentText("");
+    private final EntityPlayerMP player;
+    @Nullable private Text header;
+    @Nullable private Text footer;
+    private final Map<UUID, TabListEntry> entries = Maps.newHashMap();
+
+    public SpongeTabList(EntityPlayerMP player) {
+        this.player = player;
+    }
+
+    @Override
+    public Optional<Text> getHeader() {
+        return Optional.ofNullable(this.header);
+    }
+
+    @Override
+    public TabList setHeader(@Nullable Text header) {
+        this.header = header;
+
+        this.refreshClientHeaderFooter();
+
+        return this;
+    }
+
+    @Override
+    public Optional<Text> getFooter() {
+        return Optional.ofNullable(this.footer);
+    }
+
+    @Override
+    public TabList setFooter(@Nullable Text footer) {
+        this.footer = footer;
+
+        this.refreshClientHeaderFooter();
+
+        return this;
+    }
+
+    @Override
+    public TabList setHeaderAndFooter(@Nullable Text header, @Nullable Text footer) {
+        // Do not call the methods, set directly
+        this.header = header;
+        this.footer = footer;
+
+        this.refreshClientHeaderFooter();
+
+        return this;
+    }
+
+    private void refreshClientHeaderFooter() {
+        S47PacketPlayerListHeaderFooter packet = new S47PacketPlayerListHeaderFooter();
```

Comment:
There is a constructor for the header, then you don't need an AT for the header field.